[DOCS] Add GUI for Azure/AWS/GCP Secret Engines configuration (#29647)
hashicorp/vault: website/content/docs/secrets/azure.mdx

<Note>

You can configure the Azure secrets engine with the Vault API or
established environment variables such as `AZURE_CLIENT_ID` or
`AZURE_CLIENT_SECRET`. If you use both methods, note that
environment variables always take precedence over API values.

</Note>

Most secrets engines must be configured in advance before they can perform their
functions. These steps are usually completed by an operator or configuration
management tool.

Enable the Azure secrets engine:

<Tabs>

> <Tab heading="CLI">

Reviewer: ```suggestion
<Tab heading="CLI"  group="cli">
```

```shell-session
$ vault secrets enable azure
Success! Enabled the azure secrets engine at: azure/
```

By default, the secrets engine will mount at the name of the engine. To
enable the secrets engine at a different path, use the `-path` argument.

</Tab>

> <Tab heading="GUI">

Reviewer: ```suggestion
<Tab heading="GUI"  group="gui">
```

![Partial screenshot of the Vault GUI showing the "Identity token key" search select input](/img/gui/azure/identityTokenKey.png)

You must have `list` permission on the `identity/oidc/key` endpoint to view
existing workload identity federation (WIF) identity token keys during setup.

1. Open the web UI for your Vault instance.
1. Login under the target namespace or choose the target namespace from the
   selector at the bottom of the left-hand menu and re-authenticate.
1. Select **Secrets Engines** from the left-hand menu.
1. Click **Enable new engine +** on the **Secrets Engines** page.
1. Select **Azure**.
1. Click **Next**.
1. Set the mount path for the Azure plugin. For example, `azure`.
1. If you use WIF, add the identity token key:
   - Click **Method Options**.
   - Click **Identity Token Key**.
   - Enter your new key name or select one from the token key list.
1. Click **Enable engine**.
1. Click **Save** to enable the plugin.

</Tab>

</Tabs>

Configure the secrets engine with account credentials:

<Tabs>

> <Tab heading="CLI">

Reviewer: ```suggestion
<Tab heading="CLI"  group="cli">
```

```shell-session
$ vault write azure/config \
    subscription_id=$AZURE_SUBSCRIPTION_ID \
    tenant_id=$AZURE_TENANT_ID \
    client_id=$AZURE_CLIENT_ID \
    client_secret=$AZURE_CLIENT_SECRET

Success! Data written to: azure/config
```

</Tab>

> <Tab heading="GUI">

Reviewer: ```suggestion
<Tab heading="GUI"  group="gui">
```

1. Login under the target namespace or choose the target namespace from the
   selector at the bottom of the left-hand menu and re-authenticate.
1. Select **Secrets Engines** from the left-hand menu.
1. Select your `azure` plugin you want to update.
1. Click **Configure**.
1. Enter your configuration information.
1. Set the access type <EnterpriseAlert inline="true" />
1. Save your changes.

</Tab>

</Tabs>

If you are running Vault inside an Azure VM with MSI enabled, `client_id` and `client_secret` may be omitted. For more information on authentication, see the [authentication](#authentication) section below.

In some cases, you cannot set sensitive account credentials in your Vault configuration. For example, your organization may require that all security credentials are short-lived or explicitly tied to a machine identity.

To provide managed identity security credentials to Vault, we recommend using Vault
[plugin workload identity federation](#plugin-workload-identity-federation-wif)
(WIF) as shown below.

Alternatively, configure the audience claim value and the Client, Tenant and Subscription IDs for plugin workload identity federation:

<Tabs>

> <Tab heading="Vault CLI">

Reviewer: ```suggestion
<Tab heading="Vault CLI"  group="cli">
```

```shell-session
$ vault write azure/config \
    subscription_id=$AZURE_SUBSCRIPTION_ID \
    tenant_id=$AZURE_TENANT_ID \
    client_id=$AZURE_CLIENT_ID \
    identity_token_audience=$TOKEN_AUDIENCE
```

</Tab>

> <Tab heading="GUI">

Reviewer: ```suggestion
<Tab heading="GUI"  group="gui">
```

1. Enable the Azure secrets engine:

<Tabs>

> <Tab heading="Vault CLI">

Reviewer: ```suggestion
<Tab heading="CLI">
```

   ```shell-session
   $ vault secrets enable azure
   Success! Enabled the azure secrets engine at: azure/
   ```

   By default, the secrets engine will mount at the name of the engine. To
   enable the secrets engine at a different path, use the `-path` argument.

</Tab>

> <Tab heading="Vault UI">

Reviewer: ```suggestion
<Tab heading="GUI">
```

> 

Reviewer: We should use screenshots sparingly. Generally, we want to pick one "hero" screenshot that helps folks orient themselves in the GUI to make sure they're looking at the right screen.

> 1. Open the web UI for your Vault instance.
> 1. Select **Secret Engines** from the left-hand menu.
> 1. Right click **Enable new engine** on the Secret Engines page. 
> 1. Select **Azure**.
> 1. Click **Next**.
> 1. Set the mount path for the Azure plugin. For example, `azure`.
> 1. If you are an Enterprise user configuring Workload Identity Federation, you can add the Identity Token Key during the mounting process through the UI. 
>    - Click **Method Options**.
>    - Click on the **Identity Token Key** input. If you have **list** permissions to the `identity/oidc/key` you will see a dropdown of available keys. You can create your own if one does not exists. 
>    ![Partial screenshot of the Vault GUI showing the "Identity token key" search select input](/img/gui/azure/identityTokenKey.png)
>    - If you do not have **list** permissions to `identity/oidc/key` you can input your own key name.
>    ![Partial screenshot of the Vault GUI showing the "Identity token key" input when user does not have list permissions to `identity/oidc/key`](/img/gui/azure/noList.png)
> 
> 1. Click **Enable engine**.
> 1. Click **Save** to enable the plugin.

Reviewer: ```suggestion
![Partial screenshot of the Vault GUI showing the "Identity token key" search select input](/img/gui/azure/identityTokenKey.png)

You must have `list` permission on the `identity/oidc/key` endpoint to view
existing workload identity federation (WIF) identity token keys during setup.

1. Open the web UI for your Vault instance.
1. Login under the target namespace or choose the target namespace from the
   selector at the bottom of the left-hand menu and re-authenticate.
1. Select **Secrets Engines** from the left-hand menu.
1. Click **Enable new engine +** on the **Secrets Engines** page.
1. Select **Azure**.
1. Click **Next**.
1. Set the mount path for the Azure plugin. For example, `azure`.
1. If you use WIF, add the identity token key:
    - Click **Method Options**.
    - Click **Identity Token Key**. 
    - Enter your new key name or select one from the token key list.
1. Click **Enable engine**.
1. Click **Save** to enable the plugin.
```

In general, we want to keep how-to guides simple by focusing on the optimal path and simplifying the instructions.

</Tab>

</Tabs>

1. Configure the secrets engine with account credentials:

<Tabs>

> <Tab heading="Vault CLI">

Reviewer: ```suggestion
<Tab heading="CLI">
```

   Success! Data written to: azure/config
   ```
</Tab>

> <Tab heading="Vault UI">

Reviewer: ```suggestion
<Tab heading="GUI">
```

> 1. Click **Configure** from the Azure secret engine configuration view.
> ![Partial screenshot of the Vault GUI showing the "Configure" Toolbar option](/img/gui/azure/configure-prompt.png)
> 1. On the configuration form enter your configuration information. If you are an Enterprise user, you will see the option to select Access Type.
> ![Partial screenshot of the Vault GUI showing the Configuration form](/img/gui/azure/configure.png)
> 1. Save your changes.

Reviewer: ```suggestion
![Partial screenshot of the Vault GUI showing the Configuration form](/img/gui/azure/configure.png)

1. Open the web UI for your Vault instance.
1. Login under the target namespace or choose the target namespace from the
   selector at the bottom of the left-hand menu and re-authenticate.
1. Select **Secrets Engines** from the left-hand menu.
1. Select your `azure` plugin you want to update.
1. Click **Configure**.
1. Enter your configuration information.
1. Set the access type <EnterpriseAlert inline="true">
1. Save your changes.
```

</Tab>

</Tabs>

If you are running Vault inside an Azure VM with MSI enabled, `client_id` and `client_secret` may be omitted. For more information on authentication, see the [authentication](#authentication) section below.

In some cases, you cannot set sensitive account credentials in your Vault configuration. For example, your organization may require that all security credentials are short-lived or explicitly tied to a machine identity.

To provide managed identity security credentials to Vault, we recommend using Vault
[plugin workload identity federation](#plugin-workload-identity-federation-wif)
(WIF) as shown below.

> <Note> Workload identity federation is only available with Vault Enterprise. </Note>

Reviewer: ```suggestion
```
We should use the "enterprise-only" or "enterprise-and-hcp" partials at the top of the relevant section/page rather than adding an adhoc alert here.

Alternatively, configure the audience claim value and the Client, Tenant and Subscription IDs for plugin workload identity federation:

<Tabs>

<Tab heading="Vault CLI">

   ```shell-session
   $ vault write azure/config \
       subscription_id=$AZURE_SUBSCRIPTION_ID \
       tenant_id=$AZURE_TENANT_ID \
       client_id=$AZURE_CLIENT_ID \
       identity_token_audience=$TOKEN_AUDIENCE
   ```

</Tab>

> <Tab heading="Vault UI">

Reviewer: ```suggestion
<Tab heading="GUI">
```
